### PR TITLE
Convert photo picker fragment to kotlin to avoid memory leaks

### PR DESCRIPTION
**Pull request by planarvoid**
Keeping views as fields leads to memory leaks. This PR fixes this issue by converting this the PhotoPickerFragment to Kotlin and replacing the fields with Kotlin synthetic accessors. As an additional advantage it will be much easier to use MVVM while working on the media picker consolidation project.
The changes should be straightforward without changing any logic.
Note: The code contains a solution to keep position in the list on orientation changes. This doesn't seem to work. I've checked the play store version and it doesn't work there either so I guess we'll fix this in the next phase of this project.

To test:
- Try adding a video/image block in GB editor
- Click on "Choose from device"
- Check that everything works as expected 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.kt [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+                container,
+                false
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        if (browserType.isWPStoriesPicker) {
+            wp_stories_take_picture.visibility = View.VISIBLE
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.kt:135]: no action required, just sharing 😊: I'm always a bit in doubt on which notation to use when using kotlin synthetic accessors. Found sometimes useful (to keep xml ids convention in XML only) to use the `as` like

```
import kotlinx.android.synthetic.main.photo_picker_fragment.wp_stories_take_picture as wpStoriesTakePicture
```

of course this make you lose the concise

```
import kotlinx.android.synthetic.main.photo_picker_fragment.*
```
but on the other hand it keeps code naming conventions 🤷‍♂️.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -961,12 +961,12 @@ public void onPhotoPickerHidden() {
...
         }
     }
-
+    
     /*
      * called by PhotoPickerFragment when media is selected - may be a single item or a list of items
      */
     @Override
-    public void onPhotoPickerMediaChosen(@NonNull final List<Uri> uriList) {
+    public void onPhotoPickerMediaChosen(@NotNull List<? extends Uri> uriList) {
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:969]: I think I removed this by mistake, good catch 👍

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.kt (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.kt:135]: this is an interesting point 👍 . I don't love the `*` import but I couldn't make it work with directly importing `import kotlinx.android.synthetic.main.photo_picker_fragment.wp_stories_take_picture` (AS was fighting with me). The alias seems like a useful feature. I'll try it and let you know :-)

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -290,7 +290,7 @@
     <string name="button_sync">Sync</string>
     <string name="button_view">View</string>
     <string name="button_preview">Preview</string>
-    <string name="button_insert">Insert</string>
+    <string name="button_insert" tools:ignore="UnusedResources">Insert</string>
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/res/values/strings.xml:293]: great catch! It was caused by AS replacing `R.string.button_insert` by `string.button_insert` which is ignored by lint. I've fixed it in all the places where I used `UnusedResources` ignore before

**Comment by planarvoid** [comment]
Thanks for the review @develric ! I've addressed your concerns. Let me know what you think 👍

**Pull request merged**